backend/src/test/java/org/cryptomator/hub/license/LicenseHolderTest.java

```diff
@@ -157,6 +168,122 @@ public void testSetInvalidToken() {
 		}
 	}
 
+	@Nested
+	@TestProfile(LicenseHolderInitPropsTest.ValidInitPropsInstanceTestProfile.class)
+	@DisplayName("Testing LicenseHolder methods using InitProps")
+	class LicenseHolderInitPropsTest {
+
+		@InjectMock
+		Session session;
+
+		@InjectMock
+		LicenseValidator validator;
+
+		MockedStatic<Settings> settingsClass;
+
+		public static class ValidInitPropsInstanceTestProfile implements QuarkusTestProfile {
+			@Override
+			public Map<String, String> getConfigOverrides() {
+				return Map.of("hub.initial-id", "42", "hub.initial-license", "token");
+			}
+		}
+
+		@BeforeEach
+		public void setup() {
+			Query mockQuery = Mockito.mock(Query.class);
+			Mockito.doNothing().when(session).persist(Mockito.any());
+			Mockito.when(session.createQuery(Mockito.anyString())).thenReturn(mockQuery);
+			Mockito.when(mockQuery.getSingleResult()).thenReturn(0l);
+
+			settingsClass = Mockito.mockStatic(Settings.class);
+		}
+
+		@AfterEach
+		public void teardown() {
+			settingsClass.close();
+			Arc.container().instance(LicenseHolder.class).destroy();
+		}
+
+		@Test
+		@DisplayName("If init token is valid, set it in license holder")
+		public void testValidInitTokenSet() {
+			var decodedJWT = Mockito.mock(DecodedJWT.class);
+			Mockito.when(validator.validate("token", "42")).thenReturn(decodedJWT);
+			Settings settingsMock = new Settings();
+			settingsMock.hubId = "42";
+			settingsClass.when(Settings::get).thenReturn(settingsMock);
+
+			holder.init();
+
+			Mockito.verify(validator, Mockito.times(2)).validate("token", "42");
+			Assertions.assertEquals(decodedJWT, holder.get());
+		}
+
+		@Test
+		@DisplayName("If init token is invalid, do net set it in license holder and nullify db entry")
+		public void testInitTokenOnFailedValidationNotSet() {
+			Mockito.when(validator.validate("token", "42")).thenAnswer(invocationOnMock -> {
+				throw new JWTVerificationException("");
+			});
+			Settings settingsMock = new Settings();
+			settingsMock.hubId = "42";
+			settingsClass.when(Settings::get).thenReturn(settingsMock);
+
+			holder.init();
+
+			Mockito.verify(validator, Mockito.times(2)).validate("token", "42");
+			Mockito.verify(session, Mockito.times(2)).persist(Mockito.eq(settingsMock));
+			Assertions.assertNull(holder.get());
+		}
+
+		@Test
+		@DisplayName("Setting a valid token validates and overwrites the init token")
+		public void testSetValidToken() {
+			var decodedJWT = Mockito.mock(DecodedJWT.class);
+			Mockito.when(validator.validate("token3000", "42")).thenReturn(decodedJWT);
+
+			Settings initSettingsMock = new Settings();
+			initSettingsMock.licenseKey = "token";
+			initSettingsMock.hubId = "42";
+			settingsClass.when(Settings::get).thenReturn(initSettingsMock);
```

Comment:
What is this test prepartion used for? I cannot see its relevance in the test.

Reply:
This got simplified in [76c57bc](https://github.com/cryptomator/hub/pull/229/commits/76c57bcc74945c88c2cb6c171a8f94d0ace7b6c9). The idea is now to only test if setting the token overwrites the init token retrieved from the properties. Before, I wanted to include that a licence is also read from the database, but this does not change anything regarding the setting of the licence, but worsens the readability.